lib/parsers/asm-parser-dotnet.ts

```diff
@@ -74,13 +76,21 @@ export class DotNetAsmParser implements IAsmParser {
                 usedLabels.push(labelResult.value[1]);
             }
 
-            let methodResult = trimmedLine.matchAll(methodRefRe).next();
-            if (methodResult.done) methodResult = trimmedLine.matchAll(tailCallRe).next();
+            const methodResult = trimmedLine.matchAll(methodRefRe).next();
             if (!methodResult.done) {
-                const name = methodResult.value[1];
+                let name = methodResult.value[2];
+                if (name.startsWith('[')) {
```

Comment:
For my own understanding this change is to change what the `name` is considered to be in `jmp [rax]Foo`; ie handling some kind of offset-indirect addressing?

Before, as I understand it:
- the regex matches _anything_ in `[]` after a single word and returns the contents of the `[]` , so `blahblah [yibble]boing` retured just `yibble`

Now:
- the regex only matches `call` or `jump` or `tail.jmp` then _everything_ to EOL
- then the code here picks out the `[` and... if it's exactly `[moo]` it will substring the `moo`
- else it _includes_ the literal `[]` in the output so `[rax]Foo` becomes `[rax]`
- if it doesn't contain a `[` the whole string is returned, so `jmp wombat` returns `wombat`

Am I reading that right?  I initially thought "surely there's a regex to do all this" but I couldn't write one if so...but it does seem a shame to mix regexes with all this `includes`/substringing stuff (which is supposedly regex's sweet spot)...

Posting this anyway in case my thoughts/confusion above inspires you to think a little more about a better way to do this.

Reply:
Almost right modulus:
* else it _includes_ the literal `[]` in the output so `[rax]Foo` becomes `Foo`

Reply:
Actually there's yet another case `call [rax][Foo]` and we need the `Foo` here, but I cannot think any way other than matching `[` twice.

Reply:
> `[rax]Foo` becomes `Foo`

 _might_ be regexable. But honestly, golfing this stuff isn't really the goal here. Just making sure I understand it enough to say "seems ok", which it does. If/when you update for the second case, any consideration to simplifying is apprecitaed.

Reply:
For now I'm going to leave this for future improvement.